### ISSUE TITLE
Slash inserter: VoiceOver doesn't read the correct menu option names.

### ISSUE THREAD

**Issue by kjellr** [WordPress/gutenberg#8462]
**Describe the bug**

VoiceOver's narration of the items in the `/` inserter is often out of sync with the actual selection. 

**To Reproduce**

1. Activate VoiceOver
2. In a new line, type `/` to open the slash inserter.
3. Navigate down through the list, and check to see if the selected item matches the spoken text.

**Expected behavior**

I expected VoiceOver to read the correct block names.

**Actual behavior**

When you first hit `/`, VoiceOver lets you know there are 10 results, but it does not alert you to the currently selected option. As you begin to navigate down using the arrow keys, VoiceOver often says the incorrect block name. 

**Screenshots**

![voiceover](https://user-images.githubusercontent.com/1202812/43659741-edf08ef0-972a-11e8-8d5a-79b45687ce85.gif)

**Desktop**
 - OS: Mac OS 10.13.6
 - Safari 11.1.2
 - Gutenberg 3.4

**Comment by afercia** [comment]
Noting there's also a core Trac ticket about this issue, where "sometimes the wrong option gets announced". See https://core.trac.wordpress.org/ticket/38637

**Comment by timwright12** [comment]
I'm happy to look at this one, but I can't for the life of me find it in the file system. can someone point me to the code?

**Comment by timwright12** [comment]
@kjellr @afercia Looks like this may have been fixed in Safari. Tested in v12.01 and could not replicate. If I had to guess I'd say that this was caused by the initial focus of the popover being set to the first item instead of the container, so there was an offset happening.

Ran the test three times with the same results.

![ezgif-1-f94e40f74a7f](https://user-images.githubusercontent.com/459581/49668163-50be8100-fa2b-11e8-9cac-f620bf468761.gif)

**Comment by afercia** [comment]
@timwright12 Thanks for looking into this. Yep, seems to me this is something that regressed at some point in Safari / VoiceOver and it relates to all the ARIA combobox implementations, not only the Gutenberg inserter.

As mentioned in https://core.trac.wordpress.org/ticket/38637 the classic editor Tags autocompleter used to work in macOS El Capitan + Safari 9. Then stopped working properly starting from macOS Sierra + Safari 10.  Also the W3C examples linked in the Trac ticket didn't work.

Testing both in Gutenberg (e.g. slash inserter, @ mentions, Tags suggestions) and in the classic editor (Tags suggestions) seems everything works as expected in Safari 12 / VoiceOver 9.

It would be great to test again with Safari 11, to confirm it's a user agent bug. Maybe WCUS contributor day would be a good opportunity to make a call for someone who's still on Sierra 🙂 

If confirmed, I'd tend to think there's no much we can do. For the future, I'd suggest to change the auto-completers to meet the ARIA 1.1 implementation, which is slightly different from the ARIA 1.0 one (links in the Trac ticket) and seemed to work better with Safari 11. Of course, we should make sure the new ARIA 1.1 implementation is supported by all the main browsers / screen readers combinations. That would require a new round of testing and I'd say it should preferably be addressed in a separate issue.

**Comment by afercia** [comment]
See related #12254

**Comment by paaljoachim** [comment]
@joedolson @kjellr 
Can you recheck with the current version of WordPress and Gutenberg?
Thanks!

**Comment by paaljoachim** [comment]
I just tested on my Mac running OSX 10.15.7
WordPress 5.7 RC1
Twenty Twenty One

I heard the correct voiceover and saw the correct labels.

https://user-images.githubusercontent.com/5323259/109213779-4d9d4000-77b1-11eb-9318-dfada771d562.mp4

**Comment by Mamaduka** [comment]
@paaljoachim, @kjellr, should we close this issue?

**Comment by carolinan** [comment]
I tried this on Safari 16.5.2 with VoiceOver, and VoiceOver did not read out any block names.
It reads out the instructions for the slash inserter, and then it announces that you are inside a text area:
(turn the sound on if possible)

https://github.com/WordPress/gutenberg/assets/7422055/22414035-9262-45e7-998a-d019a2bf8278


It works well with Chrome and VoiceOver on macOs.

**Comment by t-hamano** [comment]
Is this problem still occurring in Safari? I'm on a Windows machine, so I can't test it in Safari, but it seems to work fine in Chrome/Firefox.

- OS: Windows 11
- Browser: Chrome 120.0.6099.109, Firefox 120.0.1
- Screen reader: NVDA 2023.2.0.3096

https://github.com/WordPress/gutenberg/assets/54422211/6c2a7f53-df45-44b1-9252-56a4af65b517

**Comment by talldan** [comment]
> Is this problem still occurring in Safari?

It's better, the items are announced, but it doesn't announce the currently selected item when you first type a letter or as you continue typing 🤔 

It looks like https://github.com/WordPress/gutenberg/pull/54902 fixed some issues related to this, and that it stopped working was down to Safari being buggy.

**Comment by Mamaduka** [comment]
I can confirm that the issue still exists with VoiceOver and Safari 18.4. The initially selected item isn't announced.

**Comment by Mamaduka** [comment]
I just retested this with the latest Safari (Version 26.1) and VoiceOver, and the first selected option is announced correctly.

It seems the problem was fixed in recent Safari versions. I'm going to close this. We can open a new issue if this regresses again.

## Screencast

https://github.com/user-attachments/assets/2712b8f5-7027-4ccb-9ce7-86f6af3ed5fa
